plugin/builtin/patchtransformer/PatchTransformer_test.go

```diff
@@ -746,8 +743,72 @@ spec:
       - image: test-image
         name: test-deployment
         ports:
+        - containerPort: 8080
+          name: take-over-the-world
+          protocol: TCP
         - containerPort: 8080
           name: disappearing-act
           protocol: TCP
 `)
 }
+
+func TestPatchTransformerSecondaryMergeKey(t *testing.T) {
```

Comment:
I don't know how the coverage exactly is here, but I would expect more tests to make sure that this does the right thing.